application/models/Survey.php

```diff
@@ -2351,7 +2357,8 @@ public function getSurveyUrl($language = null, $params = [], $preferShortUrl = t
 
         // If short url is not preferred or no alias is found, return a traditional URL
         $urlParams = array_merge($params, ['sid' => $this->sid, 'lang' => $language]);
-        $url = Yii::app()->createAbsoluteUrl('survey/index', $urlParams);
+        $url = App()->getController()->createAbsoluteUrl('survey/index', $urlParams);
```

Comment:
I don't like to call a controller from a model, but if no code reorg is to be done, I don't see any other way

Reply:
Maybe you can just move the code to a helper?
Maybe now is a good time to create App::createPublicUrl() ?

Reply:
ok, you convinced me now :)

Reply:
Please check my latest commit